### PR TITLE
Get uploads in once

### PR DESCRIPTION
**Pull request by DinhHuy2010**
Adding functions to get uploads in utils.py for less use quota since only requests once.

# Related Issue/Addition to code

- While fetching block of videos if will requests playlist ID every single time
- So if requests a lot of videos, if will be requests every time block
- So only requests in once and that it

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Proposed Changes
- Fix operations.py in recent videos retrieve function
- Add functions to utils.py

# Additional Info
- You can fix in official release.

## Checklist:

- [x] My code follows the style guidelines of this project and have read CONTRIBUTING.md
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

# Screenshots

Original | Updated
:----------------------:|:-----------:
no response  | Screenshot 2022-04-11 171757

### CODE REVIEW

File: Scripts/utils.py [hunk 1]
```diff
@@ -213,3 +213,15 @@ def print_error_title_fetch():
   print(f"  > You won't be able to delete/hide any comments like usual, but you can {F.LIGHTMAGENTA_EX}exclude users before saving the log file{S.R}")
   print(f"  > Then, you can {F.LIGHTGREEN_EX}delete the comments later{S.R} using the {F.YELLOW}mode that removes comments using a pre-existing log file{S.R}")
   input("\n Press Enter to continue...")
+
+
+# Fetch the uploads playlist ID for the channel
+# By DinhHuy2010 at GitHub
```
Comment (Firecul) [inline Scripts/utils.py:219]: I feel this line isn't necessary 
```suggestion

```

Comment (DinhHuy2010) [reply Scripts/utils.py:219]: ok, you can remove it

**Review by TechStudent11 — CHANGES_REQUESTED** [review]

File: Scripts/utils.py [hunk 2]
```diff
@@ -213,3 +213,15 @@ def print_error_title_fetch():
...
+
+def get_uploads_playlist_id(channel_id):
+  try:
+    channel = auth.YOUTUBE.channels.list(part="contentDetails", id=channel_id).execute()
+    uploadplaylistId = channel["items"][0]["contentDetails"]["relatedPlaylists"]["uploads"]
+  except HttpError:
+    uploadplaylistId = "[Unavailable]" # If raise HttpError, set to [Unavailable]
```
[the remark this answers is not in the document]
Comment (DinhHuy2010) [reply Scripts/utils.py:225]: yes

File: Scripts/operations.py [hunk 3]
```diff
@@ -1381,7 +1374,7 @@ def get_block_of_videos(nextPageToken, j, k, numVideosBlock = 50):
 
     for item in result['items']:
       videoID = str(item['snippet']['resourceId']['videoId'])
-      videoTitle = str(item['snippet']['title']).replace("&quot;", "\"").replace("&#39;", "'")
+      videoTitle = str(item['snippet']['title']) # Video title been HTML unescaped. So no need to replace.
```
[the remark this answers is not in the document]
Comment (DinhHuy2010) [reply Scripts/operations.py:1377]: uh what?

[the remark this answers is not in the document]
Comment (TechStudent11) [reply Scripts/operations.py:1377]: You changed a part of the code, then proceeded to say that it does one thing when it actually doesn't. 

TL;DR this should be a totally different PR.

[the remark this answers is not in the document]
Comment (EthanHindmarsh) [reply Scripts/operations.py:1377]: Not sure what you're saying TechStudent11🤔
They removed an unnecessary replace, and put a comment in about it

[the remark this answers is not in the document]
Comment (TechStudent11) [reply Scripts/operations.py:1377]: oops. once again I miss-read something. the var name is `videoTitle` in a function about videos, exactly what this PR is about. Sorry!

**Comment by DinhHuy2010** [comment]
hello?

**Comment by Masamune3210** [comment]
....Don't do that, the project will get messed with when they have a chance, they do have other things in their life you know

**Comment by KendallDoesCoding** [comment]
> Reference in

Yeah, have patience @DinhHuy2010

**Comment by TechStudent11** [comment]
> > Reference in
> 
> Yeah, have patience @DinhHuy2010

That's ironic.

To explain further; you comment on issues and PRs asking "Any update on this?". Not making fun, just pointing that out.

**Comment by KendallDoesCoding** [comment]
> > > Reference in
> > 
> > 
> > Yeah, have patience @DinhHuy2010
> 
> That's ironic.
> 
> To explain further; you comment on issues and PRs asking "Any update on this?". Not making fun, just pointing that out.

True, yeah I do that but usually I try to put one comment as a reminder and if I get no reply, I try my best not to put any more comments unless it's very urgent cause I know yall are busy with school, work, etc unless if it's really urgent and needs to get done ASAP.

**Comment by DinhHuy2010** [comment]
ok

**Comment by DinhHuy2010** [comment]
4tgfkrjgifjrigfnrj4gbjfierg

**Comment by KendallDoesCoding** [comment]
> 4tgfkrjgifjrigfnrj4gbjfierg

?????????

**Comment by Masamune3210** [comment]
I will get responded to once they have a chance, please don't spam post

**Comment by DinhHuy2010** [comment]
any updates??

**Comment by KendallDoesCoding** [comment]
nop

File: YTSpammerPurge.py [hunk 4]
```diff
@@ -673,16 +676,16 @@ def primaryInstance(miscData):
...
             if config['skip_confirm_video'] == False:
               if i == 10 and len(videosToScan) > 11:
                 remainingCount = str(len(videosToScan) - 10)
                 userChoice = choice(f"There are {remainingCount} more recent videos, do you want to see the rest?")
                 if userChoice == False:
                   break
                 elif userChoice == None:
-                  return True # Return to main menu
-            print(f"  {i+1}. {videosToScan[i]['videoTitle']}")
+                  return True # Return to main menu
```
Comment (Firecul) [inline YTSpammerPurge.py:687]: Absolutely a tiny thing (and not important at all,) my OCD compels me to point out the extra spaces added here.

Comment (DinhHuy2010) [reply YTSpammerPurge.py:687]: ok???

Comment (TechStudent11) [reply YTSpammerPurge.py:687]: > Absolutely a tiny thing (and not important at all,) my OCD compels me to point out the extra spaces added here.

I mean, he removed it from the code, so I guess it's fine.

Comment (Firecul) [reply YTSpammerPurge.py:687]: Not these ones @TechStudent11 
image

Comment (TechStudent11) [reply YTSpammerPurge.py:687]: > Not these ones @TechStudent11 image

OH GOSH WHY

Comment (DinhHuy2010) [reply YTSpammerPurge.py:687]: ???

Comment (DinhHuy2010) [reply YTSpammerPurge.py:687]: updates?

Comment (DinhHuy2010) [reply YTSpammerPurge.py:687]: updates?

Comment (DinhHuy2010) [reply YTSpammerPurge.py:687]: updates?

Comment (Firecul) [reply YTSpammerPurge.py:687]: He is clearly busy.

Comment (DinhHuy2010) [reply YTSpammerPurge.py:687]: updates???????????

Comment (TechStudent10) [reply YTSpammerPurge.py:687]: > He is clearly busy.

@DinhHuy2010

Comment (KendallDoesCoding) [reply YTSpammerPurge.py:687]: @DinhHuy2010 the more you ask for updates, the longer it will take to be reviewed by @ThioJoe :) , don't be impatient, some PR's I've seen are open for 2-5 years, so ThioJoe will surely review when he got time man

Comment (DinhHuy2010) [reply YTSpammerPurge.py:687]: ok

Comment (DinhHuy2010) [reply YTSpammerPurge.py:687]: it since 1 week and 5 days

Comment (KendallDoesCoding) [reply YTSpammerPurge.py:687]: > it since 1 week and 5 days

bro, just stop. 
or
@ThioJoe will just not review this PR and close it. He will have a look when he's free. It's his project.

Comment (KendallDoesCoding) [reply YTSpammerPurge.py:687]: Plus, as per your username, you are born in **2010**, which means you are 12 years OLD. 

You require to be at least 13+ to be using GitHub or any social media.

Comment (TechStudent10) [reply YTSpammerPurge.py:687]: I don't think having a year in your username means that year is your year of birth. Could just be a very important year to them.

**Comment by KendallDoesCoding** [comment]
Fair enough, but he acts like a 12 year old by giving repeated reminders
and crap, if you get what I mean.
